File: frontend/src/types/movement.ts
export type MovementType = 'receipt' | 'expenditure' | 'write_off'

export const MOVEMENT_LABELS: Record<MovementType, string> = {
  receipt: 'Приход',
  expenditure: 'Расход',
  write_off: 'Списание',
}

export const MOVEMENT_COLORS: Record<MovementType, string> = {
  receipt: 'green',
  expenditure: 'red',
  write_off: 'gray',
}

export interface Movement {
  id: string
  company_id: string
  entity_id: string
  record_id: string
  movement_type: MovementType
  quantity: number
  unit: string | null
  notes: string | null
  reference_number: string | null
  performed_by: string | null
  performed_at: string
  created_at: string
}

export interface MovementCreate {
  movement_type: MovementType
  quantity: number
  unit?: string
  notes?: string
  reference_number?: string
  performed_at?: string
}

export interface MovementBalance {
  receipt: number
  expenditure: number
  write_off: number
  balance: number
  unit: string | null
}

export interface PaginatedMovements {
  items: Movement[]
  total: number
  page: number
  pages: number
  size: number
}

File: backend/app/schemas/movement.py
import uuid
from datetime import datetime
from typing import Literal
from pydantic import BaseModel, Field


MovementTypeStr = Literal["receipt", "expenditure", "write_off"]


class MovementCreate(BaseModel):
    movement_type: MovementTypeStr
    quantity: float = Field(gt=0)
    unit: str | None = None
    notes: str | None = None
    reference_number: str | None = None
    performed_at: datetime | None = None


class MovementResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    company_id: uuid.UUID
    entity_id: uuid.UUID
    record_id: uuid.UUID
    movement_type: str
    quantity: float
    unit: str | None
    notes: str | None
    reference_number: str | None
    performed_by: uuid.UUID | None
    performed_at: datetime
    created_at: datetime


class MovementBalance(BaseModel):
    receipt: float
    expenditure: float
    write_off: float
    balance: float
    unit: str | None
